src/SensioLabs/Melody/Handler/AuthenticableHandlerInterface.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+namespace SensioLabs\Melody\Handler;
+
+use SensioLabs\Melody\Configuration\UserConfiguration;
+use SensioLabs\Melody\Exception\InvalidCredentialsException;
+
+/**
+ * Interface AuthenticableHandlerInterface.
+ *
+ * @author Maxime STEINHAUSSER <maxime.steinhausser@gmail.com>
+ */
+interface AuthenticableHandlerInterface extends ResourceHandlerInterface
+{
+    const CREDENTIALS_NORMAL = 'normal';
+    const CREDENTIALS_SECRET = 'secret';
+
+    /**
+     * Returns an array of the required credentials.
+     *
+     * @return array
+     */
+    public function getRequiredCredentials();
+
+    /**
+     * Provides credentials to be used at some point by the handler.
+     *
+     * @param array             $credentials
+     * @param UserConfiguration $userConfig
+     *
+     * @throws InvalidCredentialsException
+     */
+    public function provideCredentials(array $credentials, UserConfiguration $userConfig);
```

Comment:
1. `setCredentials`.
2. it's not the responsibility of the `AuthenticableHandlerInterface` to implement the save of credentials in the user config